presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -269,4 +273,14 @@ public final void close()
             throw Throwables.propagate(e);
         }
     }
+
+    private void cancelAllQueries()
+    {
+        QueryManager queryManager = coordinator.getQueryManager();
+        for (QueryInfo queryInfo : queryManager.getAllQueryInfo()) {
+            if (!queryInfo.getState().isDone()) {
+                queryManager.cancelQuery(queryInfo.getQueryId());
+            }
+        }
```

Comment:
The state is changed synchronously while the actual cancellation is asynchronous. Changed the order of server creation to be on the safer side. But didn't add the loop as suggested.